doc/release/upcoming_changes/12065.performance.rst

```diff
@@ -0,0 +1,8 @@
+Faster version of ``np.isin`` and ``np.in1d`` for integer arrays
+----------------------------------------------------------------
+``np.in1d`` (used by ``np.isin``) can now switch to a faster algorithm
+(up to >10x faster) when it is fed two integer arrays.
+The algorithm bares similarities to a counting sort in that it
```

Comment:
This compares to a counting sort, but the docstring compares to a radix sort. I don't have any formal CS background so either or both can be correct, just pointing this out in case you want to make the same comparison in both places.

Reply:
I think from the description, counting sort is closer, but I'd be careful with that approach as if you feed it the max value of `uint64`, it could OOM, which leads me to believe that the actual implementation is similar to a radix sort.

Reply:
The current "new" implementation seems to explicitly allocate an auxiliary array of size `arr2.ptp()` (give or take), so the potential OOM issue is probably present.

Reply:
> actual implementation is similar to a radix sort.

I just checked the implementation: It's a counting sort, with an automatic switch-over.

Reply:
Fixed the docstring, thanks.